app/src/main/java/com/example/simplemp3/ui/activity/MainActivity.kt

```diff
@@ -1,13 +1,120 @@
 package com.example.simplemp3.ui.activity
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.ComponentName
+import android.content.ServiceConnection
 import android.os.Bundle
+import android.os.Handler
+import android.os.IBinder
+import android.widget.SeekBar
+import androidx.appcompat.app.AppCompatActivity
 import com.example.simplemp3.R
+import com.example.simplemp3.service.PlayService
+import com.example.simplemp3.service.PlayService.Companion.DEFAULT_LOOP_STATE
+import com.example.simplemp3.service.PlayService.Companion.getPlayerIntent
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
+    companion object {
+        const val DELAY_SEEK_TIME = 100L
+        const val SONG_NAME = "Ao Moi Ca Mau"
+        const val ARTIST_NAME = "Jang Mi"
+
+        fun timerFromDuration(duration: Int): String {
+            val durationInSecond = duration / 1000
+            val minuteString = String.format("%02d", durationInSecond / 60)
+            val secondString = String.format("%02d", durationInSecond % 60)
+            return "$minuteString:$secondString"
+        }
+    }
+
+    private var isServiceConnected = false
+    private var isPlaying = false
+    private var isLooping = DEFAULT_LOOP_STATE
+    private var playService: PlayService? = null
+    private val serviceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            val binder = service as PlayService.PlayerServiceBinder
+            playService = binder.getService()
+            isServiceConnected = true
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) {
+            isServiceConnected = false
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        bindService(getPlayerIntent(this), serviceConnection, BIND_AUTO_CREATE)
+        displaySongDetails()
+        initTimeLoader()
+        initListener()
+    }
+
+    private fun displaySongDetails(songName: String = SONG_NAME, artistName: String = ARTIST_NAME) {
+        text_song_name.text = songName
+        text_artist_name.text = artistName
+    }
+
+    private fun initTimeLoader() {
+        Handler().apply {
+            post(object : Runnable {
+                override fun run() {
+                    bar_current_position.progress = playService?.getCurrentPercent() ?: 0
+                    text_current_timer.text = getCurrentTimerString()
+                    text_total_timer.text = getTotalTimerString()
+                    this@apply.postDelayed(this, DELAY_SEEK_TIME)
+                }
+            })
+        }
+    }
+
+    private fun initListener() {
+        image_play_pause.setOnClickListener {
```

Comment:
lần sau e có thể đặt tên ntn cho xml
```suggestion
        imageSwitchState.setOnClickListener {
```

Reply:
Đặt camelCase cho xml là chỉ cho id của view item hay cho cả các color, dimen... ạ ?